lib/exec.test.js

```diff
@@ -0,0 +1,63 @@
+const fs = require('fs')
+const path = require('path')
+
+const exec = require('./exec').exec
+
+describe('exec', () => {
+  it('should be able to create and delete files', async () => {
+    const fileName = 'this-is-a-test-file.txt'
+    const filePath = path.join(__dirname, fileName)
+
+    expect(fs.existsSync(filePath)).toBe(false)
+
+    await exec(`touch ${fileName}`, { cwd: __dirname })
+    expect(fs.existsSync(filePath)).toBe(true)
+
+    await exec(`rm ${fileName}`, { cwd: __dirname })
+    expect(fs.existsSync(filePath)).toBe(false)
+  })
+
+  it('should provide an output from stdout', async () => {
+    const outputParts = []
+
+    await exec('echo hello', {}, output => outputParts.push(output.toString()))
+
+    expect(outputParts[0].trim()).toEqual('hello')
+  })
+  // it('should error when a command fails', async () => {
+  //   let catchCalled = false
+  //
+  //   await exec('sadflkajsdfoiewflkjsadf')
+  //     .catch(() => {
+  //       catchCalled = true
+  //     })
+  //
+  //   expect(catchCalled).toBe(true)
+  // })
+  // it('should provide the error code and message', async () => {
+  //   let catchCalled = false
+  //
+  //   await exec('which asldfkj')
+  //     .catch((e) => {
+  //       catchCalled = true
+  //       expect(e.code).toBe(1)
+  //       expect(e.message).toBe('Exit code was 1')
+  //       expect(e.errorOutput).not.toBeDefined()
+  //     })
+  //
+  //   expect(catchCalled).toBe(true)
+  // })
+  // it('should provide the error code and message', async () => {
+  //   let catchCalled = false
+  //
+  //   await exec('echo >&2 "this is the error message"; exit 4')
+  //     .catch((e) => {
+  //       catchCalled = true
+  //       expect(e.code).toBe(4)
+  //       expect(e.message).toBe('Exit code was 4')
+  //       expect(e.errorOutput).toBe('this is the error message\n')
+  //     })
+  //
+  //   expect(catchCalled).toBe(true)
+  // })
```

Comment:
I'm guessing this commented code was committed accidentally?

Reply:
This test works on my mac but not on the Windows agents, the parts of the new functionality I'm using worked in the Windows test I did with @joelanman so I'm seeing it as something to investigate later (feel free to question that).  I left the test commented out so that it's there to look at later, I can push it to a different branch if that's preferable but I thought it was more likely to get missed if I did that.

I can also add a comment to explain why it's commented out.

Reply:
Ah cool, that makes sense. Yeah a comment would be good, you might always consider using [`test.skip`](https://jestjs.io/docs/api#testskipname-fn), or the [`testSkipFailingIf` construct](https://jestjs.io/docs/api#testskipname-fn), so that indentation and syntax highlighting are kept.